src/chapters/unit-testing/testing-in-csharp.rst

```diff
@@ -0,0 +1,115 @@
+
+Refactoring
+^^^^^^^^^^^
+
+When we **refactor** code, we rewrite without adding new features. Refactoring can 
+increase efficiency at runtime, but it may also mean inadvertently introducing some bugs in the process.
+Unit tests verify the most basic functionality of your code, thus safeguarding against 
+bugs introduced in refactoring. 
+
+Imagine this common workflow: 
+
+#. You practice **test-driven development (TDD)**, writing your tests to stipulate your class's code. 
+
+#. You write your class's code to pass they tests. 
+
+#. Later, a stakeholder in the project requests that you refactor your code using 
+   different syntax.
+
+The features will be the same, but the implementation changes. 
+Because the changes in implementation do not require changes to the outcomes, 
+unit tests will help in this scenario. If your tests continue
+to pass after the refactor, you can move on, knowing you have not 
+inadvertently introduced a bug. Writing tests just once provides innumerable 
+benefits for the whole lifetime of the codebase.
+
+Documentation
+^^^^^^^^^^^^^
+
+Unit tests serve as key documentation for fellow programmers. Again, 
+because they address the most fundamental tasks of your classes,
+unit tests serve as live-code use-cases. You may also have an 
+external documentation directory with examples of how to run your
+project, or perhaps you have been writing comments within your code
+to best communicate with your teammates about your changes. Both of
+these are great choices and should be done when possible. However, they 
+also require more forethought to maintain. Each time you update
+your code, you might not remember to update the documentation and 
+comments. With unit testing, however, you have a more obvious reminder
+that a change has been made if a test fails.
+
+.. _testing-best-practices:
+
+Testing Best Practices
+----------------------
+
+Below are some best practices to keep in mind when writing unit tests, in any language.
+
+#. The AAAs
+
+   The AAAs of unit testing refers to the pattern to follow when 
+   writing your unit tests. 
+
+   a. Arrange the variables your test requires
+   b. Act on the methods your test requires
+   c. Assert the anticipated comparison of the expected and actual values
+
+#. Deterministic
+
+   *Every* time a test is run, it should produce the same outcome. 
+   A test that passes only most of the time is a worthless test.
+
+#. Relevant
+
+   Group tests by related class and function.
+
+#. Meaningful
+
+   There is no need to test trivial code. For example, unless a getter or setter contains some 
+   additional functionality, you do not need to write a test for those methods. 
+
+Check Your Understanding
+------------------------
+
+.. admonition:: Question
+
+   True or False: Comments are the best tool to make your code readable.
+
+.. ans: False, comments are helpful but can be used in tandem with other forms of documentation, including unit tests
```

Comment:
The section establishes that comments and unit tests are both vital and important. I think this question should be changed.

Reply:
I disagree, the text is meant to position unit tests as a superior form of documentation. It describes comments as flawed, pointing out that they require forethought to maintain and can become out of sync with the code, which is less likely to happen with unit testing. I just changed the descriptor in the first sentence of that section to help drive home this point

Reply:
That's fair. I was reading it more from the perspective of someone who has heard over the past couple of months that comments are also vital, so as a student, I may find comments the best tool to make code readable. I think if you wanted to eliminate some of that subjectivity, you could do a multiple choice question such as: "What form of documentation requires the least amount of forethought when it comes to maintenance?"